lib/plugins/acts_as_customizable/lib/acts_as_customizable.rb

```diff
@@ -155,7 +155,7 @@ def respond_to?(method, include_private = false)
         def for_custom_field_accessor(method_symbol)
           match = /\Acustom_field_(?<id>\d+)=?\z/.match(method_symbol.to_s)
           if match
-            custom_field = CustomField.find_by_id(match[:id])
+            custom_field = available_custom_fields.find { |cf| cf.id.to_s == match[:id] }
```

Comment:
Beware that this little beast will not always return a Scope.

See the `WorkPackage`, as far as I can tell it creates a list/array/whatever-not-a-scope, which is cached:
````ruby
#work_package.rb
def available_custom_fields
  (project && type) ? (project.all_work_package_custom_fields & type.custom_fields.all) : []
end

# project.rb
def all_work_package_custom_fields(options = {})
  @all_work_package_custom_fields ||= (
    WorkPackageCustomField.for_all(options) + (
      if options.include? :include
        WorkPackageCustomField.joins(:projects)
          .where(projects: { id: id })
          .includes(options[:include]) # use #preload instead of #includes if join gets too big
      else
        work_package_custom_fields
      end
    )
  ).uniq.sort
end
````

Thus I think `find_by_id` will not work.
So you are right, performance will be best when the caller asks for more than one custom field, because in that case `project.all_work_package_custom_fields` will have been cached.

Do I have any better options performance-wise?